src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotDefineFinalizersForTypesDerivedFromMemoryManagerTests.cs

```diff
@@ -0,0 +1,229 @@
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+            Imports System
+            Imports System.Buffers
+
+            Namespace TestNamespace
+                MustInherit Class TestClass(Of T)
+                    Inherits MemoryManager(Of T)
+
+                    Public Overrides Function Pin(ByVal Optional elementIndex As Integer = 0) As MemoryHandle
+                        Throw New NotImplementedException()
+                    End Function
+
+                    Public Overrides Sub Unpin()
+                    End Sub
+
+                    Protected Overrides Sub Finalize()
+                    End Sub
+
+                    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
+                    End Sub
+                End Class
+            End Namespace",
+                GetVisualBasicWarningResultAt(16, 45));
+        }
+
+        [Fact]
+        public async Task ClassIndirectlyDerivedFromMemoryManagerWithFinilizerWarns()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+            using System;
+            using System.Buffers;
+
+             namespace TestNamespace
+             {
+                class Deeper<T> : Middle<T>
+                {
+                    ~Deeper() { }
+                }
+
+                class Middle<T> : MemoryManager<T>
+                {
+                    public override Span<T> GetSpan()
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override MemoryHandle Pin(int elementIndex = 0)
+                    {
+                        throw new NotImplementedException();
+                    }
+
+                    public override void Unpin() { }
+
+                    protected override void Dispose(bool disposing) { }
+                }
+            }",
+                GetCSharpWarningResultAt(9, 22));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+            Imports System
+            Imports System.Buffers
+
+            Namespace TestNamespace
+                    MustInherit Class Deeper(Of T)
+                    Inherits Middle(Of T)
+
+                    Protected Overrides Sub Finalize()
+                    End Sub
+                End Class
+
+                MustInherit Class Middle(Of T)
+                    Inherits MemoryManager(Of T)
+
+                    Public Overrides Function Pin(ByVal Optional elementIndex As Integer = 0) As MemoryHandle
+                        Throw New NotImplementedException()
+                    End Function
+
+                    Public Overrides Sub Unpin()
+                    End Sub
+
+                    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
+                    End Sub
+                End Class
+            End Namespace",
+                GetVisualBasicWarningResultAt(9, 45));
+        }
+
+        private DiagnosticResult GetCSharpWarningResultAt(int line, int column)
```

Comment:
💡 In general, new tests do not need the `GetCSharpWarningResultAt` and `GetVisualBasicWarningResultAt` methods. The contents can be inlined for the rare case where this syntax is used.